buildSrc/src/main/groovy/net/minecraftforge/forge/tasks/Util.groovy

```diff
@@ -47,100 +55,117 @@ public class Util {
-	}
-
-	public static def getArtifacts(project, config, classifiers) {
-		def ret = [:]
-		config.resolvedConfiguration.resolvedArtifacts.each {
-			def art = [
-				group: it.moduleVersion.id.group,
-				name: it.moduleVersion.id.name,
-				version: it.moduleVersion.id.version,
-				classifier: it.classifier,
-				extension: it.extension,
-				file: it.file
-			]
-			def key = art.group + ':' + art.name
-			def folder = "${art.group.replace('.', '/')}/${art.name}/${art.version}/"
-			def filename = "${art.name}-${art.version}"
-			if (art.classifier != null)
-				filename += "-${art.classifier}"
-			filename += ".${art.extension}"
-			def path = "${folder}${filename}"
-			def url = "https://libraries.minecraft.net/${path}"
-			if (!checkExists(url)) {
-				url = "https://maven.minecraftforge.net/${path}"
-			}
-			ret[key] = [
-				name: "${art.group}:${art.name}:${art.version}" + (art.classifier == null ? '' : ":${art.classifier}") + (art.extension == 'jar' ? '' : "@${art.extension}"),
-				downloads: [
-					artifact: [
-						path: path,
-						url: url,
-						sha1: sha1(art.file),
-						size: art.file.length()
-					]
-				]
-			]
-		}
-		return ret
-	}
+    }
 
-    public static def getMavenPath(task) {
-        def classifier = task.archiveClassifier.get()
-        def dep = "${task.project.group}:${task.project.name}:${task.project.version}" + (classifier == '' ? '' : ':' + classifier)
-        return "${task.project.group.replace('.', '/')}/${task.project.name}/${task.project.version}/${task.project.name}-${task.project.version}".toString() + (classifier == '' ? '' : '-' + classifier) + '.jar'
+    public static String[] getClasspath(project, libs, artifact) {
+        def ret = []
+        artifactTree(project, artifact).each { key, lib ->
+            libs[lib.name] = lib
+            if (lib.name != artifact)
+                ret.add(lib.name)
+        }
+        return ret
     }
 
-    public static def getMavenDep(task) {
-        def classifier = task.archiveClassifier.get()
-        return "${task.project.group}:${task.project.name}:${task.project.version}" + (classifier == '' ? '' : ':' + classifier)
+    public static def getArtifacts(project, config) {
+        def ret = [:]
+        config.resolvedConfiguration.resolvedArtifacts.each { dep ->
+            def info = getMavenInfoFromDep(dep)
+            def url = "https://libraries.minecraft.net/$info.path"
+            if (!checkExists(url))
+                url = "https://maven.minecraftforge.net/$info.path"
+                
+            ret[info.key] = [
+                name: info.name,
+                downloads: [
+                    artifact: [
+                        path: info.path,
+                        url: url,
+                        sha1: dep.file.sha1(),
+                        size: dep.file.length()
+                    ]
+                ]
+            ]
+        }
+        return ret
+    }
+    
+    public static def getMavenInfoFromDep(dep) {
```

Comment:
```suggestion
    public static Map getMavenInfoFromDep(dep) {
```
Adding explicit types to dynamic Groovy code improves IDE support as well as performance a bit